libexec/revng/revng-check-conventions

```diff
@@ -1,498 +1,727 @@
-            | cat_header "Includes should never use <..> except for C++ standard includes:"
-    else
-        $GREP "^\s*#include <.*\.hpp>" "${C_FILES[@]}" | \
-            cat_header "Includes should never use <..> except for C++ standard includes:"
-    fi
-
-    # Parenthesis at the end of line (except for raw strings)
-    $GREP "(\$" "${C_FILES[@]}" | grep -v 'R"LLVM.*(' | cat_header "Parenthesis at the end of line:"
-
-    # Things should never be at the beginning of a line
-    REGEXPS=('\*>' '/[^/\*]' ':[^:\(]*)' '==' '\!=' '<[^<]' '>' '>=' '<=' '//\s*WIP' '#if\s*[01]')
-    for REGEXP in "${REGEXPS[@]}"; do
-        $GREP "^\s*$REGEXP" "${C_FILES[@]}"
-    done | cat_header "Found snippets that should never be at the beginning of a line:"
-
-    # Check there are no static functions in C++ header files
-    for FILE in "${CXX_FILES[@]}"; do
-        if [[ $FILE == *h ]]; then
-            $GREP -H '^static\b[^=]*$' "$FILE" | cat
-            head -n1 "$FILE" |  grep -E '^#pragma once$' > /dev/null || \
-                echo "$FILE: header does not start with #pragma once"
-        fi
-    done
-}
+    parser.add_argument(
+        "--print-timings",
+        action="store_true",
+        help="After execution, print timing information for each pass",
+    )
 
-# Run revng-specific checks on all files
-function run_revng_checks() {
-    # Ban whitespaces at the end of a line
-    $GREP " \$" "${ALL_FILES[@]}" | cat_header "Whitespace at the end of line:"
+    parser.add_argument(
+        "--use-local-clang-format-file",
+        action="store_true",
+        help="""Instead of using the repository config, let clang-format search in its default
+locations for a config file.
+See clang-format documentation for more details.""",
+    )
 
-    # Ban tabs altogether
-    $GREP -P "\t" "${ALL_FILES[@]}" | cat_header "Tabs present:"
+    parser.add_argument(
+        "--print-clang-format-config",
+        action="store_true",
+        help="""Print the clang-format configuration to stdout and then exit, ignoring other
+arguments except for --use-local-clang-format-file.""",
+    )
 
-    # Ensure all files end with a newline
-    for FILE in "${ALL_FILES[@]}"; do
-        if [[ -s "$FILE" && ! $(tail -c 1 "$FILE" | base64) == "Cg==" ]]; then
-            echo "$FILE"
-        fi
-    done | cat_header "Files that don't end in a newline:"
-}
+    parser_file_selection = parser.add_mutually_exclusive_group()
+    parser_file_selection.add_argument(
+        "--commit-range", metavar="RANGE", help="Check files changed in the specified commit range"
+    )
+    parser_file_selection.add_argument(
+        "--HEAD", action="store_true", help="Alias for '--commit-range HEAD^..HEAD'"
+    )
+    parser_file_selection.add_argument(
+        "--cached", action="store_true", help="Only check staged files"
+    )
+    parser_file_selection.add_argument(
+        "--all", action="store_true", help="Check all files tracked by git"
+    )
```

Comment:
`--all` absolutely needs to be the default, in particular now that we're super-parallel. It has been a constant source of mistakes.
We can still benefit `--HEAD` in `orc check-branch`!

Reply:
removed `--all`, it is now the default if no files are provided